### PR TITLE
Add Heat plugin

### PR DESCRIPTION
**Pull request by dragorosson**

### CODE REVIEW

**Comment by lekhajee** [comment]
@dragorosson : Thank you for your contribution! Looks like travis is failing due to lint and for py26 support.

I am labeling this as wip, please feel free to change the label to ready for review when its ready. 
Also, which project is this for, just so we can label it.

**Comment by ksheedlo** [comment]
#318

**Comment by dragorosson** [comment]
Thanks, @lekhajee! I've updated the PR so that it passes. Please let me know what should be changed to adhere to best practices.

**Comment by lekhajee** [comment]
@dragorosson : thanks. I am assuming this is ready for review and labelling it.

File: mimic/model/heat_objects.py [hunk 1]
```diff
@@ -0,0 +1,232 @@
...
+@stack_deletion.declare_default_behavior
+def default_delete_behavior(collection, request, stack_name, stack_id):
+    """
+    Successfully delete a stack as long as it exists. Updates the stacks status.
+    """
+    stack = collection.stack_by_id(stack_id)
+
+    if not stack:
+        request.setResponseCode(404)
+        return ''
```
Comment (lekhajee) [inline mimic/model/heat_objects.py]: Can we change all the no content responses to be bytes instead i.e. `b''`

File: mimic/model/heat_objects.py [hunk 2]
```diff
@@ -0,0 +1,232 @@
...
+        """
+        for stack in self.stacks:
+            if stack.stack_id == stack_id:
+                return stack
+
+    def request_list(self, show_deleted=False, tags=[]):
+        """
+        Tries a stack list operation.
+        """
+        def should_show_stack(stack):
```
Comment (lekhajee) [inline mimic/model/heat_objects.py]: missing docstring

File: mimic/rest/heat_api.py [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+    def _region_collection_for_tenant(self, tenant_id):
+        return (self._api_mock._get_session(self._session_store, tenant_id)
+                .collection_for_region(self._region_name))
+
+    app = MimicApp()
+
+    @app.route('/v1/<string:tenant_id>/stacks', methods=['POST'])
+    def create_stack(self, request, tenant_id):
+        """
+        Creates a stack.
```
Comment (lekhajee) [inline mimic/rest/heat_api.py]: Can we add links to the docs for all the API calls

File: mimic/model/heat_objects.py [hunk 4]
```diff
@@ -0,0 +1,232 @@
...
+        return stack
+
+    def creation_response_json(self):
+        """
+        Returns the response associated with the stack's creation.
+        """
+        return {
+            "stack": {
+                "id": self.stack_id
+            }
```
Comment (lekhajee) [inline mimic/model/heat_objects.py]: looks like this response is missing `links`

File: mimic/model/heat_objects.py [hunk 5]
```diff
@@ -0,0 +1,232 @@
...
+
+    def json(self):
+        """
+        Returns the JSON representation of the stack.
+        """
+        return {
+            'stack_name': self.stack_name,
+            'stack_status': self.action + '_' + self.status,
+            'id': self.stack_id,
+            'tags': ','.join(self.tags)
```
Comment (lekhajee) [inline mimic/model/heat_objects.py]: This list response seems different from what I see in http://docs.rackspace.com/orchestration/api/v1/orchestration-devguide/content/GET_stack_list__v1__tenant_id__stacks_Stack_Operations.html.

File: mimic/model/heat_objects.py (hunk 1 above)
Comment (dragorosson) [inline mimic/model/heat_objects.py]: Done, for both occurrences.

File: mimic/model/heat_objects.py (hunk 2 above)
Comment (dragorosson) [inline mimic/model/heat_objects.py]: Done, and added docstring to (hopefully) all method definitions, including the test file.

File: mimic/rest/heat_api.py (hunk 3 above)
Comment (dragorosson) [inline mimic/rest/heat_api.py]: Done, for all 4 endpoints.

File: mimic/model/heat_objects.py (hunk 4 above)
Comment (dragorosson) [inline mimic/model/heat_objects.py]: Added `links` to both `creation_response_json` and `json`.

File: mimic/model/heat_objects.py (hunk 5 above)
Comment (dragorosson) [inline mimic/model/heat_objects.py]: I have added the missing keys in, but their values are "Not implemented". If this is not sufficient for now, let me know.

**Comment by lekhajee** [comment]
+1, thanks for doing this!

**Pull request merged**